### PR TITLE
Move runtime into modules

### PR DESCRIPTION
**Pull request by arv**
This moves the traceur runtime into modules inside runtime/modules.
This is in preparation for allowing importing the runtime as
modules in the future.

### CODE REVIEW

**Comment by arv** [comment]
This is step towards importing the runtime instead of using `$traceurRuntime`. Once this lands and we have pushed to npm it will be possible to replace `traceurRuntime.foo` with `require('traceur/dist/commonjs/runtime/modules/foo.js')`.

File: Makefile [hunk 1]
```diff
@@ -302,7 +302,7 @@ updateSemver: # unless the package.json has been manually edited.
 dist/commonjs: bin/traceur.js
 	./traceur --dir src/ dist/commonjs/ --modules=commonjs
 
-prepublish: bin/traceur.js bin/traceur-runtime.js
+prepublish: bin/traceur.js bin/traceur-runtime.js dist/commonjs/
```
Comment (johnjbarton) [inline Makefile]: I guess this is a convenience for you, it's not something we use in the make or npm builds right?

**Comment by johnjbarton** [comment]
LGTM

You might consider rebuilding the commit with `git mv` from the `/runtime` to `/modules` directory and then adding the `/runtime` files as new as this would make the blame history more useful.

File: Makefile (hunk 1 above)
Comment (arv) [inline Makefile]: Removing.

**Comment by arv** [comment]
`git log --stat -C -M` show the moves/copies. Another option would be to do two commits but I don't think that is what you meant.

**Pull request merged**